mllib/src/main/scala/org/apache/spark/ml/clustering/PowerIterationClustering.scala

```diff
@@ -166,6 +166,7 @@ class PowerIterationClustering private[clustering] (
     val w = if (!isDefined(weightCol) || $(weightCol).isEmpty) {
       lit(1.0)
     } else {
+      SchemaUtils.checkColumnTypes(dataset.schema, $(weightCol), Seq(FloatType, DoubleType))
```

Comment:
shall we check if it is a `NumericType`? An integer column with value 1 is a valid input as well and if someone is using it, this may introduce a regression.